minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h

```diff
@@ -407,8 +409,9 @@ RotationQuaternionTemplate<Scalar>::operator*(
 
   // check if the multiplication has resulted in the quaternion no longer being
   // approximately normalized.
-  if (std::abs(result.squaredNorm() - static_cast<Scalar>(1.0)) >
-      EPS<Scalar>::normalization_value()) {
```

Comment:
The problem can occur with both doubles and floats. 

It shows up pretty reliably when integrating large numbers of angular velocities to get orientations. Basically as each multiplication gives a result whose quaternion is not exactly equal to 1 if you perform enough multiplications to a quaternion its square norm sort of goes on a random walk until by chance it has changed enough to fall the check.